lewis_emulators/linmot/device.py

```diff
@@ -0,0 +1,103 @@
+from collections import OrderedDict
+
+from states import WarnStateCode, ErrorStateCode
+
+from states import StoppedState, MovingState
+from lewis.devices import StateMachineDevice
+
+HARD_LIMIT_MINIMUM = 0.0
+HARD_LIMIT_MAXIMUM = 5000.0
+
+# Defaults taken from device
+DEVICE_DEFAULT_VELO = 52
+DEVICE_DEFAULT_MAX_ACCEL = 10
+DEVICE_DEFAULT_SPEED_RES = 190735
+
+states = OrderedDict([("Stopped", StoppedState()),
+                      ("Moving", MovingState())])
+
+
+class SimulatedLinmot(StateMachineDevice):
+
+    inside_hard_limits = False
+
+    def _initialize_data(self):
+        """
+        Initialize all of the device's attributes.
+        """
+        self.position = 0
+        self.target_position = 0
+        self.inside_hard_limits = True
+
+        self.velocity = DEVICE_DEFAULT_VELO
+        self.maximal_acceleration = DEVICE_DEFAULT_MAX_ACCEL
+        self.speed_resolution = DEVICE_DEFAULT_SPEED_RES
+
+        self.motor_warn_status = WarnStateCode.STATIONARY
+        self.motor_error_status = ErrorStateCode.NONE
+
+        self.new_action = False
+        self.position_reached = False
+        self.tolerance = 0.01
+
+        self.connected = True
+
+    def _get_transition_handlers(self):
+        return OrderedDict([
+            (("Stopped", "Moving"), lambda: self.new_action is True and self.position_reached is False),
+            (("Moving", "Stopped"), lambda: self.position_reached is True),
+        ])
+
+    @property
+    def state(self):
+        return self._csm.state
+
+    @property
+    def device_error(self):
+        """
+        Is the device errored due to being outside of the hard limits
+
+        Return(s):
+            (bool): True if device in errored state
+        """
+        return True if not self.within_hard_limits() else False
```

Comment:
Can be shortened thus:
```suggestion
        return not self.within_hard_limits()
```